src/services/dragonchain-client/DragonchainClient.ts

```diff
@@ -70,368 +67,720 @@ export class DragonchainClient {
+     *
+     * If nothing is provided, it will list at the root of the heap
+     * @example folder1
+     * @example folder1/subFolder
+     */
+    prefixKey?: string,
+    /**
+     * Smart contract to list the objects from
+     *
+     * When running from within a smart contract, this is provided via the SMART_CONTRACT_ID environment variable, and doesn't need to be explicitly provided
+     */
+    smartContractId?: string
+  }) => {
+    if (!options.smartContractId) {
+      if (!process.env.SMART_CONTRACT_ID) throw new FailureByDesign('BAD_REQUEST', 'Parameter `smartContractId` is required when not running within a smart contract')
+      options.smartContractId = process.env.SMART_CONTRACT_ID
+    }
+    let path = `/list/${options.smartContractId}/`
+    if (options.prefixKey) {
+      if (options.prefixKey.endsWith('/')) throw new FailureByDesign('BAD_REQUEST', 'Parameter `prefixKey` cannot end with \'/\'')
+      path += options.prefixKey + '/'
     }
-    return await this.get(`/verifications/${blockId}`) as Response<Verifications>
-  }
-
-  /**
-   * getSmartContractHeap
-   * Get from the smart contract heap
-   * This function, (unlike other SDK methods) returns raw utf-8 text by design.
-   * If you expect the result to be parsed json pass `true` as the jsonParse parameter.
-   * @param {string} key the key under which data has been stored in heap
-   * @param {string} scName the name of smart contract
-   * @param {boolean} jsonParse attempt to parse heap data as JSON. Throws JSONParse error if it fails.
-   */
-  public getSmartContractHeap = async (key: string, scName: string, jsonParse: boolean = false) => {
-    const response = await this.get(`/get/${scName}/${key}`, jsonParse)
-    return response as Response<string>
-  }
-
-  /**
-   * listSmartcontractHeap
-   * List objects from a smart contract heap
-   * @param {string} scName the name of smart contract
-   * @param {string} key the sub-key ('folder') to list in the SC heap (optional. Defaults to root of SC heap)
-   */
-  public listSmartcontractHeap = async (scName: string, key: string = '') => {
-    let path = `/list/${scName}/`
-    if (key) path += key
     return await this.get(path) as Response<string[]>
   }
 
   /**
-   * registerTransactionType
-   * Registers a new transaction type
-   * @param {TransactionTypeStructure} txnTypeStructure
+   * Create a new transaction type for ledgering transactions
    */
-  public registerTransactionType = async (txnTypeStructure: TransactionTypeStructure) => {
-    return await this.post('/transaction-type', txnTypeStructure) as Response<UpdateResponse>
+  public createTransactionType = async (options: {
+    /**
+     * The string of the transaction type to create
+     * @example cust1
+     */
+    transactionType: string,
+    /**
+     * The custom indexes that should be associated with this transaction type
+     */
+    customIndexes?: TransactionTypeCustomIndexes[]
```

Comment:
This type should prolly be singular, right?
```suggestion
    customIndexes?: TransactionTypeCustomIndex[]